src/Utilities.UnitTests/Options/SymbolNamesWithValueOptionTests.cs

```diff
@@ -271,46 +271,41 @@ public class MyClass {}
-        [InlineData("MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.this[]*", "this[]")]
-        [InlineData("MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyEvent*", "MyEvent")]
-        [InlineData("MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyMethod(*", "MyMethod")]
-        [InlineData("MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyMethod2(System.Str*", "MyMethod2")]
+        // Symbol name
+        [InlineData("My*", "MyCompany")]
+        [InlineData("My*", "MyOuterClass")]
+        [InlineData("My*", "MyInnerClass")]
+        [InlineData("My*", "MyField")]
+        [InlineData("My*", "MyProperty")]
+        [InlineData("My*", "MyEvent")]
+        [InlineData("My*", "MyMethod")]
+        [InlineData("My*", "MyMethod2")]
         // Fully qualified name with prefix
         [InlineData("N:MyCompany*", "MyCompany")]
         [InlineData("N:MyCompany.MyProduct*", "MyProduct")]
         [InlineData("N:MyCompany.MyProduct.MyFeature*", "MyFeature")]
         [InlineData("T:MyCompany.MyProduct.MyFeature.MyOuterClass*", "MyOuterClass")]
         [InlineData("T:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass*", "MyInnerClass")]
         [InlineData("F:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyField*", "MyField")]
-        [InlineData("M:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.ctor*", ".ctor")]
+        [InlineData("M:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass..ctor*", ".ctor")]
         [InlineData("P:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyProperty*", "MyProperty")]
-        [InlineData("P:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.this[]*", "this[]")]
+        [InlineData("P:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.Item*", "this[]")]
         [InlineData("E:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyEvent*", "MyEvent")]
-        [InlineData("M:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyMethod(*", "MyMethod")]
+        [InlineData("M:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyMethod*", "MyMethod")]
         [InlineData("M:MyCompany.MyProduct.MyFeature.MyOuterClass.MyInnerClass.MyMethod2(System.Str*", "MyMethod2")]
-        // Partial match
-        [InlineData("MyCompany.MyProduct*", "MyFeature")]
-        [InlineData("N:MyCompany.MyProduct*", "MyFeature")]
-        [InlineData("MyFeature*", "MyFeature")]
-        [InlineData("MyInnerClass*", "MyInnerClass")]
-        [InlineData("MyOuterClass*", "MyInnerClass")]
-        [InlineData("My*", "MyCompany")]
-        [InlineData("My*", "MyOuterClass")]
-        [InlineData("My*", "MyInnerClass")]
-        [InlineData("My*", "MyField")]
-        [InlineData("My*", "MyProperty")]
-        [InlineData("My*", "MyEvent")]
-        [InlineData("My*", "MyMethod")]
-        [InlineData("My*", "MyMethod2")]
+        // Fully qualified name without prefix
```

Comment:
I have kept these for compatibility with the old algorithm but we might want to drop them as they imply a more heavy computation.